Yoast/wordpress-seo
# P1-3 Create schema settings component

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Adds schema settings to the search appearance.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Check that the schema settings render on the Search Appearance page. The settings should be added for the posts, pages, custom post types (all 3 under the Content Types tab) and attachments (Media tab).
  * For pages, only the default page type select box should be visible.
  * For posts, custom post types and attachments the default page type and the default article type should be visible.
* The correct defaults should be (and `(default)` should be appended to that option name):
  * Post
    * Page type default: Web Page
    * Article type default: Article
  * Page/custom post type/attachments
    * Page type default: Web Page
    * Article type default: None
* Check the hidden input fields:
  * In the browser's console run `document.getElementById("hidden_schema-page-type-post").value` where the `post` part can be replaced with the page type, for example `page` or `book`.
  * The value should be the same as the value in the select box. It is slightly different than the display text. If you want to check the value itself, use `document.getElementById("schema-page-type-post").value` in the console (note it is the same code, just not prefixed with `hidden_`).
  * If you change the select box value, the hidden input value should adjust accordingly (updates after the select loses focus).
  * <img width="443" alt="console" src="https://user-images.githubusercontent.com/35524806/87415427-5a2baa00-c5cd-11ea-8435-c3e77eced7fa.png">
* Check the alert is shown when selecting any other value than previously saved.
  * <img width="622" alt="alert" src="https://user-images.githubusercontent.com/35524806/87415270-2b153880-c5cd-11ea-956d-9c11ee720070.png">
  * For pages this message should contain the singular version: `this setting` (instead of `these settings`).
* Check that values are saved properly.
  * Change some options.
  * Save changes.
  * The option should still be set as you saved it.
  * It is saved in the database in `wp_options` under `wpseo_titles` as the id field before, e.g. `schema-page-type-post` etc.
* Check the alert again. As the previously saved settings are different for the options you just saved.


## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P1-3
